### ISSUE TITLE
New rule: No return value from render

### ISSUE THREAD

**Issue by pke** [yannickcr/eslint-plugin-react#1886]
It seems there is no rule for catching missing return values from render methods:

```jsx
const Price = ({ pricings, title }) => {
  if (!(pricings.length > 0 && title)) { return }
  ...
}
```

This should be considered an error, when `undefined` is returned from a render function.

**Comment by ljharb** [comment]
Related to #1884.

This would only be able to catch cases like `return` or `return undefined`; there’s plenty of other ways you could return undefined. I’m not sure how valuable this would be as a rule.

**Comment by mzealey** [comment]
This would be very valuable as a rule. In preact `return undefined` is the same as `return null` but for react compatibility `return null` should always be used. Would help catch a lot of the common issues like this.

**Comment by ljharb** [comment]
That sounds like a good reason to have a preact-specific rule (in a preact-specific plugin).

**Comment by mzealey** [comment]
Seems more like a react-specific plugin to me as you want to ensure that there are no `return` statements without a value/with undefined in react render fn...

**Comment by ljharb** [comment]
React itself already ensures that, and if you correctly unit test everything and make warnings fail your tests, you don’t need a linter rule for it.

Explicitly catching `return` or `return undefined` might add value, but is only a fraction of the ways undefined might be returned.
